src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -0,0 +1,289 @@
+        label->setFixedWidth(label->minimumSizeHint().width());
+        font = label->font();
+        font.setBold(false);
+        label->setFont(font);
+
+        QComboBox* combo = new QComboBox(this);
+        font = combo->font();
+        font.setBold(false);
+        combo->setFont(font);
+        m_combos.append(combo);
+        combo->setModel(m_comboModel);
+        m_comboMapper->setMapping(combo, i);
+        connect(combo, SIGNAL(currentIndexChanged(int)), m_comboMapper, SLOT(map()));
+
+        //layout labels and combo fields in column-first order
+        int combo_rows = 1+(m_columnheader.count()-1)/2;
+        int x=i%combo_rows;
+        int y= 2*(i/combo_rows);
+        m_ui->gridLayout_combos->addWidget(label, x, y);
+        m_ui->gridLayout_combos->addWidget(combo, x, y+1);
+    }
+
+    m_parserModel->setHeaderLabels(m_columnheader);
+    m_ui->tableViewFields->setModel(m_parserModel);
+
+    connect(m_ui->spinBoxSkip, SIGNAL(valueChanged(int)), SLOT(skippedChanged(int)));
+    connect(m_ui->comboBoxCodec, SIGNAL(currentIndexChanged(int)), SLOT(parse()));
+    connect(m_ui->comboBoxTextQualifier, SIGNAL(currentIndexChanged(int)), SLOT(parse()));
+    connect(m_ui->comboBoxComment, SIGNAL(currentIndexChanged(int)), SLOT(parse()));
+    connect(m_ui->comboBoxFieldSeparator, SIGNAL(currentIndexChanged(int)), SLOT(parse()));
+    connect(m_ui->checkBoxBackslash, SIGNAL(toggled(bool)), SLOT(parse()));
+    connect(m_ui->pushButtonWarnings, SIGNAL(clicked()), this, SLOT(showReport()));
+    connect(m_comboMapper, SIGNAL(mapped(int)), this, SLOT(comboChanged(int)));
+
+    connect(m_ui->buttonBox, SIGNAL(accepted()), this, SLOT(writeDatabase()));
+    connect(m_ui->buttonBox, SIGNAL(rejected()), this, SLOT(reject()));
+}
+
+void CsvImportWidget::comboChanged(int comboId) {
+    QComboBox* currentSender = qobject_cast<QComboBox*>(m_comboMapper->mapping(comboId));
+    if (currentSender->currentIndex() != -1) {
+        //here is the line that actually updates the GUI table
+        m_parserModel->mapColumns(currentSender->currentIndex(), comboId);
+    }
+    updateTableview();
+}
+
+void CsvImportWidget::skippedChanged(int rows) {
+    m_parserModel->setSkippedRows(rows);
+    updateTableview();
+}
+
+CsvImportWidget::~CsvImportWidget() {}
+
+void CsvImportWidget::configParser() {
+    m_parserModel->setBackslashSyntax(m_ui->checkBoxBackslash->isChecked());
+    m_parserModel->setComment(m_ui->comboBoxComment->currentText().at(0));
+    m_parserModel->setTextQualifier(m_ui->comboBoxTextQualifier->currentText().at(0));
+    m_parserModel->setCodec(m_ui->comboBoxCodec->currentText());
+    m_parserModel->setFieldSeparator(m_ui->comboBoxFieldSeparator->currentText().at(0));
+}
+
+void CsvImportWidget::updateTableview() {
+    m_ui->tableViewFields->resizeRowsToContents();
+    m_ui->tableViewFields->resizeColumnsToContents();
+
+    for (int c=0; c<m_ui->tableViewFields->horizontalHeader()->count(); ++c) {
```

Comment:
When you're done with your changes, can you rebase your branch on develop, so we can re-review it, fix any remaining issues and then finally merge this baby?